### PR TITLE
Lb/fix sidekiq deployment

### PR DESCRIPTION
**Pull request by hut8**

### CODE REVIEW

**Comment by hut8** [comment]
Manual, one-off work just involves stopping the existing upstart script and removing the scripts in /etc/init

File: config/deploy.rb [hunk 1]
```diff
@@ -5,9 +5,10 @@
 set :repo_url, 'git@github.com:tenex/rails-assets.git'
 ask :branch, `git rev-parse --abbrev-ref HEAD`.chomp
 set :deploy_user, -> { fetch(:application) }
-set :deploy_to, lambda {
-  "/home/#{fetch :deploy_user}/rails-apps/#{fetch :application}"
-}
+set :deploy_user_home, -> { File.join '/home', fetch(:deploy_user) }
+set :rails_apps_path, -> { File.join fetch(:deploy_user_home), 'rails-apps' }
+set :deploy_to, -> { File.join fetch(:rails_apps_path), fetch(:application) }
```
Comment (joshjordan) [inline config/deploy.rb]: Are lines 9 and 10 necessary? Isn't this what `deploy_to` defaults to? Are you just trying to make sure `deploy_to` and `deploy_user_home` change in lock step?

File: config/deploy.rb [hunk 2]
```diff
@@ -40,40 +42,31 @@
...
   before :export, :upload_env do
     on roles fetch(:foreman_roles) do
-      upload! StringIO.new("RAILS_ENV=#{fetch(:stage)}"), "#{current_path}/#{foreman_env_path}"
+      upload! StringIO.new("RAILS_ENV=#{fetch(:stage)}"),
+              "#{current_path}/#{foreman_env_path}"
     end
   end
 
   after :'deploy:restart', :restart_safe do
+    invoke :'foreman:export'
```
Comment (joshjordan) [inline config/deploy.rb]: :( oops

File: config/deploy.rb (hunk 1 above)
Comment (hut8) [inline config/deploy.rb]: `deploy_to` defaults to `/var/www/rails-assets`. I'm doing this so I don't have to write out the path to home twice, once for ~/.init and once for the app itself.

File: config/deploy/production.rb [hunk 3]
```diff
@@ -2,11 +2,11 @@
...
 # Defines a single server with a list of roles and multiple properties.
 # You can define all roles on a single server, or split them:
-server 'rails-assets.tenex.tech', user: 'rails-assets', roles: %w(app db web worker)
-# server 'example.com', user: 'deploy', roles: %w{app web}, other_property: :other_value
-# server 'db.example.com', user: 'deploy', roles: %w{db}
-
-
+server(
+  'rails-assets.tenex.tech',
+  user: 'rails-assets',
```
Comment (joshjordan) [inline config/deploy/production.rb]: This seems like it should depend on a capistrano variable too

File: infrastructure/Upstart.conf [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      <allow send_destination="com.ubuntu.Upstart"
+             send_interface="org.freedesktop.DBus.Properties" />
+      <allow send_destination="com.ubuntu.Upstart"
+             send_interface="com.ubuntu.Upstart0_6" />
+      <allow send_destination="com.ubuntu.Upstart"
+             send_interface="com.ubuntu.Upstart0_6.Job" />
+      <allow send_destination="com.ubuntu.Upstart"
+             send_interface="com.ubuntu.Upstart0_6.Instance" />
+  </policy>
+</busconfig>
```
Comment (joshjordan) [inline infrastructure/Upstart.conf]: How does this differ from a default config?

Comment (joshjordan) [inline infrastructure/Upstart.conf]: Can we add some comments showing how it diverges from the default?

**Comment by joshjordan** [comment]
LGTM. Open comments but ready to merge.

File: config/deploy.rb (hunk 2 above)
Comment (hut8) [inline config/deploy.rb]: https://github.com/hyperoslo/capistrano-foreman/issues/44

File: infrastructure/Upstart.conf (hunk 4 above)
Comment (hut8) [inline infrastructure/Upstart.conf]: Done. All it does is enable user jobs.

File: config/deploy.rb (hunk 2 above)
Comment (joshjordan) [inline config/deploy.rb]: :+1:

File: infrastructure/Upstart.conf (hunk 4 above)
Comment (joshjordan) [inline infrastructure/Upstart.conf]: Thanks!

**Pull request merged**